src/commands/PublishCommand.js

```diff
@@ -151,6 +151,20 @@ export default class PublishCommand extends Command {
 
   getVersionsForUpdates(callback) {
     if (this.flags.repoVersion) {
+      // Allows automatic bumping to next semver via repoVersion flag
+      if (this.flags.repoVersion === "patch" ||
+          this.flags.repoVersion === "minor" ||
+          this.flags.repoVersion === "major" ||
+          this.flags.repoVersion === "current"
+      ) {
+        const versions = {};
+        this.updates.forEach((update) => {
+          versions[update.package.name] = this.flags.repoVersion === "current" ?
+            update.package.version : semver.inc(update.package.version, this.flags.repoVersion);
+        });
+
+        return callback(null, { versions });
+      }
```

Comment:
Hmm... actually I wonder if `--repo-version` is the right option for this.  That's really meant specifically for non-independent versioning.  And actually looking at this again it will only work correctly for independently versioned repos.  It won't update the `version` in lerna.json unless the return value from `getVersionsForUpdates` includes a `version` (singular) value.

It might actually be better to add a _new_ option that works for both independent and non-independent repos.